sycl/include/CL/sycl/multi_ptr.hpp

```diff
@@ -46,6 +46,13 @@ template <typename ElementType, access::address_space Space> class multi_ptr {
     // TODO An implementation should reject an argument if the deduced
     // address space is not compatible with Space.
   }
+#if defined(RESTRICT_WRITE_ACCESS_TO_CONSTANT_PTR)
```

Comment:
Why is this needed at all. Why not change the constructor on line 45 to from `multi_ptr(ElementType*)` to `multi_ptr(pointer_t)`. Then this extra overload isn't needed, correct?

Reply:
The spec requires both constructors, the `multi_ptr(pointer_t)` also is defined.